modules/custom-status/custom-status.php

```diff
@@ -284,14 +287,46 @@ public function load_scripts_for_block_editor() {
 
 		$publish_guard_enabled = ( 'on' === VIP_Workflow::instance()->settings->module->options->publish_guard ) ? true : false;
 
+		$custom_statuses = $this->get_custom_statuses_with_editorial_metadata();
+
 		wp_localize_script( 'vip-workflow-block-custom-status-script', 'VW_CUSTOM_STATUSES', [
 			'current_user_id'          => get_current_user_id(),
 			'is_publish_guard_enabled' => $publish_guard_enabled,
-			'status_terms'             => $this->get_custom_statuses(),
+			'status_terms'             => $custom_statuses,
 			'supported_post_types'     => VIP_Workflow::instance()->get_supported_post_types(),
 		] );
 	}
 
+	/**
+	 * Get the custom statuses, with the editorial metadatas populated under meta.
+	 *
+	 * @return array $custom_statuses The custom statuses with editorial metadatas
+	 */
+	private function get_custom_statuses_with_editorial_metadata() {
+		$editorial_metadatas = EditorialMetadata::get_editorial_metadata_terms();
+		$custom_statuses = array_map( function ( $status ) use ( $editorial_metadatas ) {
+			if ( [] !== $status->meta[ self::METADATA_REQ_EDITORIAL_IDS_KEY ] ) {
+				$meta_fields = array_values( array_filter( array_map( function ( $metadata_id ) use ( $editorial_metadatas ) {
+					// look up the metadata_id in the editorial_metadatas array, to match by term_id
+					$terms = array_values( array_filter( $editorial_metadatas, function ( $editorial_metadata ) use ( $metadata_id ) {
```

Comment:
Minor: this feels a bit awkward to search through all of `$editorial_metadatas` for each term. I also found this a bit hard to follow.

One way to handle something like this would be precomputing the mapping of metadata IDs to terms in something like `$metadata_id_to_term` computed once, and then just check the mapping in constant-time when needed instead of a second `array_filter()`. This isn't really an optimization issue as it wouldn't be a problem until there are thousands (or more) of both metadata terms and custom status associations, but I do think it could help the readability.

Reply:
Same thing as below really - this can be optimized a lot more once the refactor + bottom optimization is in. This way causes the least amount of conflicts or problems with the current code.